### ISSUE TITLE
ExplicitItLambdaParameter false positives when "it" used on multi-argument lambda

### ISSUE THREAD

**Issue by rock3r** [detekt/detekt#5108]
## Expected Behavior
The check should not flag arguments named `it` in multi-argument lambdas (you can't omit it)

## Observed Behavior
The argument gets flagged with the message ``This explicit usage of `it` as the lambda parameter name can be omitted.`` which is incorrect.

## Steps to Reproduce
Write a lambda with multiple params, for example:

```kotlin
fun myFun(lambda: (a: Int, b: Int) -> Unit) { lambda() }

myFun { a, it -> // gets flagged
  // ...
}
```

## Context
While I understand it's weird to call a multi-argument lambda parameter `it`, the suggestion in this rule is wrong. Maybe a separate rule should cover this case?

## Your Environment
* Version of detekt used: 1.21.1
* Version of Gradle used (if applicable): 7.5
* Gradle scan link (add `--scan` option when running the gradle task): not relevant
* Operating System and version: Windows 11 (not relevant)
* Link to your project (if it's a public repository): (not needed)

**Comment by cortinico** [comment]
> While I understand it's weird to call a multi-argument lambda parameter `it`,

I would argue that this is a really bad practice (we should probably have a rule about it).
But yeah this looks like a false positive and should be investigated

**Comment by BraisGabin** [comment]
Looking at [the documentation of this rule](https://detekt.dev/docs/rules/style/#explicititlambdaparameter) It seems that this is the desired behaviour:

https://github.com/detekt/detekt/blob/416efd5e5df5e2e2230d4007cc84d9c7c6bc3889/detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ExplicitItLambdaParameter.kt#L21-L28

I think that what this rule does is to ensure that no one "fools" other rules that force you to set a proper name to lambda parameters. So, to me, it should raise this case too.

**Comment by rock3r** [comment]
I'm not arguing it should not somehow be flagged — I'm arguing that it should not be flagged by this rule, or at the very least it should provide correct error message, since the current one doesn't apply to this situation (you can't just omit it)
